### DISCUSSION TITLE
.net 8.0 RID support

### DISCUSSION THREAD

**Discussion by Marek-Bor** [microsoft/WindowsAppSDK#4059] [Q&A] [answered]
Hello!

I am a civil engineer and participate in Estonian government project "e-hüppe" which translates as "e-jump" and has to do with automatization or digitalization of tasks at work. I want to know if using .NET 8.0 is a good idea (I want to create a WinUI 3 application now that will automize some routine tasks and integrate with Tekla Structures (through console .net48 IPC) as well as perform some MS office, PDF, DWG interactions). Since SDK has new RIDs (win-<arch> instead of win10-<arch>) using it creates some known issues.

I want to know if it will be a problem to implement a working app with such warnings (warning NETSDK1206: Found version-specific or distribution-specific runtime identifier(s): win10-arm64, win10-x64, win10-x86. Affected libraries: Microsoft.WindowsAppSDK) or it will work as expected? Because when I run it or test - it does not seem to fail. Also what is a roadmap for integrating these new RIDs into the SDK? Will it be done by the end of 2024 autumn? Because we need to finish this project by the end of the year and we won't get money from government if it won't be a working solution by that time. Also will turning on UseRidGraph help or it will help SDK but could cause issues in .NET 8.0?

Is using .NET 8.0 under question here and better stick to .NET 7.0 for example?

I would be very glad if someone could comment on this.

Kindest regards,
Marek Bor

**Comment by DarranRowe** [comment, marked as answer]
On UseRidGraph. The [1.4 release notes](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/stable-channel#known-issues) actually recommend it. Yes, it mentions that this is in reference to the pre-release .NET 8. This never really changed after .NET 8 was released since the servicing releases for 1.4 never added any build fixes for .NET 8.
On .NET 8 support in the Windows App SDK. The 1.5 release cycle has started. The first step of this was the release of 1.5 experimental 1. This isn't supported for production environments, but this should give an idea on when 1.5 will be released. I would say, given previous releases, that 1.5 will be released sometime in March. 1.5 experimental 1 has the build fixes for .NET 8 already, so you will see this in the final release.
There should be more than enough time to evaluate whether .NET 8 is viable.

**Comment by GreenShadeZhang** [comment]
Hello, I have upgraded .net8 in my project and published it to the app store. You can refer to my project configuration to view the main changes as shown in the figure below.

https://github.com/maker-community/ElectronBot.DotNet

![image](https://github.com/microsoft/WindowsAppSDK/assets/24240675/5411e120-7a28-4a7e-b559-3daf856da5e2)


The project can be loaded normally when opening ElectronBot.Braincase.sln.

**Reply by GreenShadeZhang** [reply]
Sorry, I misunderstood. I don't seem to understand this UseRidGraph.

https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/stable-channel#known-issues
